onnxruntime/contrib_ops/cpu/math/sparse_dense_matmul.cc

```diff
@@ -84,13 +84,37 @@ struct SparseToDenseCsr {
     const auto& b_dims = B.Shape().GetDims();
     const auto& out_dims = output.Shape().GetDims();
     auto csr_view = A.AsCsr();
-
-    ConstSparseMatrixMap<T> map_A(a_dims[0], a_dims[1], A.NumValues(),
-                                  csr_view.Outer().Data<int64_t>(),
-                                  csr_view.Inner().Data<int64_t>(),
+    const Eigen::Index* inner_index_pointer = nullptr;
+    const Eigen::Index* outer_index_pointer = nullptr;
+    std::unique_ptr<Eigen::Index[]> buffer_holder_inner, buffer_holder_outer;
+    if constexpr (std::is_same<Eigen::Index, int64_t>::value) {
```

Comment:
In general they are not the same. For this to work, we care about 3 things:
1) both types are integers
2) both are signed
3) both have the same size.

Reply:
For all the platforms we support Eigen::Index is either a 32-bit signed integer or 64-bit signed integer. No other cases.

Reply:
By default Eigen::Index is always an alias of ptrdiff_t

Reply:
What do you suggest?

Reply:
My suggestion would be essentially what I wrote in the comment
```suggestion
    if constexpr (std::is_integral<Eigen::Index>::value &&
                  std::is_signed<Eigen::Index>::value &&
                  (sizeof(Eigen::Index) == sizeof(int64_t))) {
```